hydromt/components/config.py

```diff
@@ -236,7 +234,7 @@ def create(
             {'a': 1, 'b': {'c': {'d': 2}}}
         """
         # Check if self.data is not empty
-        if len(self.data) == 0:
+        if self._data is not None and len(self._data) > 0:
```

Comment:
Good catch! But you should do the check on ``self.data`` an not ``self._data`` to force read if you have read rights here.